src/emacs/lean-info.el

```diff
@@ -67,6 +67,26 @@
       (lambda (&key record)
         (funcall cont record))))))
 
+(defun lean-get-info-record-at-point-sync ()
+  "Get info-record at point synchronously."
+  (with-demoted-errors "lean get info: %s"
+    (lean-server-send-synchronous-command
+     'info (list :file_name (buffer-file-name)
+                 :line (line-number-at-pos)
+                 :column (lean-line-offset)))))
+
+(defun lean-info-right-click-find-definition ()
+  "Offer to jump to definition of right-click target."
+  (interactive)
+  (let* ((info-record (plist-get (lean-get-info-record-at-point-sync) :record))
```

Comment:
After some more thought, I think that it would instead be best to just have a timeout. The same trick that's used to implement the synchronous Lean commands can be used to fire off a few requests for menu items, and if they don't complete in under .1 seconds, then they get ignored.

I'll try this approach.